### DISCUSSION TITLE
Help with so-elasticsearch-query

### DISCUSSION THREAD

**Discussion by newbie10122** [Security-Onion-Solutions/securityonion#6768] [Unsupported Versions] [answered]
Hello,

 I am new and have tried to read the documentation on so-elasticsearch-query. I am trying to pull out the IPs of the highly critical alerts but am not even able to simply list any records with the log level of INFO. Any assistance with my query would be appreciated. 

so-elasticsearch-query '*:so-*/_search' -d '{"query": {"match": {"level": {"query" : "INFO"}}}}'

"hits": [
      {
        "_index": "so01:so-elasticsearch-2022.01.05",
        "_type": "_doc",
        "_id": "QJmiK34Bu11ryDyZA5vA",
        "_score": 1,
        "_source": {
          "metadata": {
            "pipeline": "filebeat-7.16.2-elasticsearch-server-pipeline",
            "beat": "filebeat",
            "ip_address": "192.17.0.1",
            "type": "_doc",
            "version": "7.16.2"
          },
          "agent": {
            "hostname": "so-filebeat",
            "name": "so01",
            "id": "75d7bcc4-7e14-4b05-81af-5dbd915155e0",
            "type": "filebeat",
            "ephemeral_id": "34794877-f755-4c4b-bca2-751603dc24b3",
            "version": "7.16.2"
          },
          "log": {
            "file": {
              "path": "/logs/elasticsearch/so01.log"
            },
            "offset": 10218337,
            "level": "INFO"
          },

**Comment by dougburks** [comment]
Perhaps it is just the way that Github is rendering what you've pasted, but have you tried changing:
```
so-elasticsearch-query ':so-/_search'
```
to:
```
so-elasticsearch-query '*:so-*/_search'
```
?

(Note the asterisks.)

**Reply by newbie10122** [reply]
Thank you for the response, but yes I have it with the asterisks and it's still not returning data. Does the query seem correct or am I maybe missing a level or ??

**Reply by dougburks** [reply]
Have you tried this?
```
so-elasticsearch-query '*:so-*/_search' -d '{"query": {"match_phrase": {"log.level": "INFO"}}}'
```

**Reply by newbie10122** [reply]
Doug,

 Thank you for pointing me in the right direction. Below is the query I got to work. If anyone would like it. 

 so-elasticsearch-query '*:so-*/_search' -d '{"query": {"match_phrase": {"event.severity_label": "high"}},"size":1000}'

I do have one more question if you have an answer. I am trying to just return the source_geo.ip but cannot seem to get the syntax just right for the field. I tried the below two ways. Any help would be great.

so-elasticsearch-query '*:so-*/_search' -d '{"fields": "source_geo.ip"} {"query": {"match_phrase": {"event.severity_label": "high"}},"size":1000}' 

so-elasticsearch-query '*:so-*/_search' -d '{"query": {"match_phrase": {"event.severity_label": "high"}},"size":1000,"fields":"source_geo.ip"}'

**Reply by dougburks** [reply]
Is there a specific reason you need to query using `so-elasticsearch-query` rather than using our [Alerts](https://docs.securityonion.net/en/2.3/alerts.html) or [Hunt](https://docs.securityonion.net/en/2.3/hunt.html) interfaces which make this much easier? For example, in our Hunt interface, it should be a simple query like:
```
event.severity_label: "high" | groupby source.ip
```

If you really need to do this from the command line, one way to get a working query is to go to [Kibana](https://docs.securityonion.net/en/2.3/kibana.html) and build a visualization for the data that you want and then use the Inspect option to show the Elasticsearch request that Kibana built.

**Comment by newbie10122** [comment, marked as answer]
Doug,

 I got it to work. Thank you for the assistance. The main purpose is to
extract the source IPs and block them automatically within the firewalls.
We're just trying to leverage what you guys have already done with the
great product of Security Onion.

John
